nibabel/streamlines/trk.py

```diff
@@ -0,0 +1,713 @@
+header_2_dtd = [(Field.MAGIC_NUMBER, 'S6'),
+                (Field.DIMENSIONS, 'h', 3),
+                (Field.VOXEL_SIZES, 'f4', 3),
+                (Field.ORIGIN, 'f4', 3),
+                (Field.NB_SCALARS_PER_POINT, 'h'),
+                ('scalar_name', 'S20', MAX_NB_NAMED_SCALARS_PER_POINT),
+                (Field.NB_PROPERTIES_PER_STREAMLINE, 'h'),
+                ('property_name', 'S20',
+                 MAX_NB_NAMED_PROPERTIES_PER_STREAMLINE),
+                (Field.VOXEL_TO_RASMM, 'f4', (4, 4)),  # New in version 2.
+                ('reserved', 'S444'),
+                (Field.VOXEL_ORDER, 'S4'),
+                ('pad2', 'S4'),
+                ('image_orientation_patient', 'f4', 6),
+                ('pad1', 'S2'),
+                ('invert_x', 'S1'),
+                ('invert_y', 'S1'),
+                ('invert_z', 'S1'),
+                ('swap_xy', 'S1'),
+                ('swap_yz', 'S1'),
+                ('swap_zx', 'S1'),
+                (Field.NB_STREAMLINES, 'i4'),
+                ('version', 'i4'),
+                ('hdr_size', 'i4'),
+                ]
+
+# Full header numpy dtypes
+header_1_dtype = np.dtype(header_1_dtd)
+header_2_dtype = np.dtype(header_2_dtd)
+
+
+def get_affine_trackvis_to_rasmm(header):
+    # TRK's streamlines are in 'voxelmm' space, we will compute the
+    # affine matrix that will bring them back to RAS+ and mm space.
+    affine = np.eye(4)
+
+    # The affine matrix found in the TRK header requires the points to
+    # be in the voxel space.
+    # voxelmm -> voxel
+    scale = np.eye(4)
+    scale[range(3), range(3)] /= header[Field.VOXEL_SIZES]
+    affine = np.dot(scale, affine)
+
+    # TrackVis considers coordinate (0,0,0) to be the corner of the
+    # voxel whereas streamlines returned assumes (0,0,0) to be the
+    # center of the voxel. Thus, streamlines are shifted by half a voxel.
+    offset = np.eye(4)
+    offset[:-1, -1] -= 0.5
+    affine = np.dot(offset, affine)
+
+    # If the voxel order implied by the affine does not match the voxel
+    # order in the TRK header, change the orientation.
+    # voxel (header) -> voxel (affine)
+    header_ornt = asstr(header[Field.VOXEL_ORDER])
+    affine_ornt = "".join(aff2axcodes(header[Field.VOXEL_TO_RASMM]))
+    header_ornt = axcodes2ornt(header_ornt)
+    affine_ornt = axcodes2ornt(affine_ornt)
+    ornt = nib.orientations.ornt_transform(header_ornt, affine_ornt)
+    M = nib.orientations.inv_ornt_aff(ornt, header[Field.DIMENSIONS])
+    affine = np.dot(M, affine)
+
+    # Applied the affine found in the TRK header.
+    # voxel -> rasmm
+    voxel_to_rasmm = header[Field.VOXEL_TO_RASMM]
+    affine_voxmm_to_rasmm = np.dot(voxel_to_rasmm, affine)
+    return affine_voxmm_to_rasmm.astype(np.float32)
+
+
+def get_affine_rasmm_to_trackvis(header):
+    return np.linalg.inv(get_affine_trackvis_to_rasmm(header))
+
+
+def encode_value_in_name(value, name, max_name_len=20):
```

Comment:
Just to be sure I understand what you suggest, you would prefer to put the number of scalars associated to a property explicitly in the the name of that said property? For example, for a property named color, you would have (in the file) `color-n=3` ?

On a side note, keeping the information in ascii would consume at least 4 characters (5 for a double digit number, which is rare) of the 20 allowed for a name.

Initially, I wanted to hide that information (i.e. put it behind a \x00 EOL) because in Trackvis the property names are displayed. So a user would see `-n=3`.